crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -536,9 +544,26 @@ impl ScheduleGraph {
                             densely_chained: true
                         }
                     };
+                    let more_than_one_entry = config_iter.len() > 0;
+                    let anonymous_set = if more_than_one_entry && !collective_conditions.is_empty()
+                    {
+                        Some(AnonymousSet::new())
+                    } else {
+                        None
+                    };
+                    let prev = if let Some(anonymous_set) = anonymous_set {
```

Comment:
This repeated code can be avoided if we operate directly on the configs Vec instead of doing it in the chain code.

Reply:
Fixed in my commit below